src/OpenTelemetry/Trace/Export/Internal/BroadcastActivityProcessor.cs

```diff
@@ -0,0 +1,105 @@
+﻿// <copyright file="BroadcastActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace.Export.Internal
+{
+    internal class BroadcastActivityProcessor : ActivityProcessor, IDisposable
+    {
+        private readonly IEnumerable<ActivityProcessor> processors;
+
+        public BroadcastActivityProcessor(IEnumerable<ActivityProcessor> processors)
+        {
+            if (processors == null)
+            {
+                throw new ArgumentNullException(nameof(processors));
+            }
+
+            if (!processors.Any())
+            {
+                throw new ArgumentException($"{nameof(processors)} collection is empty");
+            }
+
+            this.processors = processors;
+        }
+
+        public override void OnEnd(Activity activity)
+        {
+            foreach (var processor in this.processors)
+            {
+                try
+                {
+                    processor.OnEnd(activity);
+                }
+                catch (Exception e)
+                {
+                    OpenTelemetrySdkEventSource.Log.SpanProcessorException("OnEnd", e);
+                }
+            }
+        }
+
+        public override void OnStart(Activity activity)
+        {
+            foreach (var processor in this.processors)
+            {
+                try
+                {
+                    processor.OnStart(activity);
+                }
+                catch (Exception e)
+                {
+                    OpenTelemetrySdkEventSource.Log.SpanProcessorException("OnStart", e);
+                }
+            }
+        }
+
+        public override Task ShutdownAsync(CancellationToken cancellationToken)
+        {
+            var tasks = new List<Task>();
+            foreach (var processor in this.processors)
+            {
+                tasks.Add(processor.ShutdownAsync(cancellationToken));
+            }
+
+            return Task.WhenAll(tasks);
```

Comment:
The `cancellationToken` is passed to all `ShutdownAsync` but if one misbehaves this line can wait forever. Is this relying on cancellation by the caller?

Reply:
The current code is vulnerable to one misbehaving processor. Not just in shutdown but even in start/stop as well. We need to address it properly.